repondeur/zam_repondeur/fetch/__init__.py

```diff
@@ -82,16 +77,6 @@ def add_article_contents_to_amendements(lecture: Lecture, articles: List[dict])
         if "alineas" in article_content and article_content["alineas"]:
             article_num, article_mult = get_article_num_mult(article_content)
             section_title = get_section_title(articles, article_content)
-            DBSession.query(AmendementModel).filter(
-                AmendementModel.chambre == lecture.chambre,
-                AmendementModel.session == lecture.session,
-                AmendementModel.num_texte == lecture.num_texte,
-                AmendementModel.organe == lecture.organe,
-                AmendementModel.subdiv_num == article_num,
-                AmendementModel.subdiv_mult == article_mult,
-            ).update(
-                {
-                    "subdiv_titre": section_title,
-                    "subdiv_contenu": article_content["alineas"],
-                }
-            )
+            DBSession.query(Article).filter(
+                Article.num == article_num, Article.mult == article_mult
+            ).update({"titre": section_title, "contenu": article_content["alineas"]})
```

Comment:
Looks like this will update all matching articles in every Lecture.

Maybe we could add `lecture_pk` in the `articles` table, with a `UniqueConstraint` on (`lecture_pk`, `num`, `mult`) ?